app/code/Magento/InventorySales/Plugin/InventoryApi/StockRepository/SaveSalesChannelsLinksPlugin.php

```diff
@@ -57,9 +82,32 @@ public function afterSave(
     ): int {
         $extensionAttributes = $stock->getExtensionAttributes();
         $salesChannels = $extensionAttributes->getSalesChannels();
+        $unAssignedWebsiteCodes = $this->getUnassignedWebsiteCodesForStock($stock);
+
         if (null !== $salesChannels) {
             try {
                 $this->replaceSalesChannelsOnStock->execute($salesChannels, $stockId);
+
+                if (count($unAssignedWebsiteCodes)) {
+                    $mergedSalesChannels = array_merge(
+                        $unAssignedWebsiteCodes,
+                        $this->getAssignedToDefaultWebsiteCodes()
+                    );
+
+                    foreach ($mergedSalesChannels as $salesChannelCode) {
```

Comment:
Why do we need to recreate SalesChannelInterface if these sales channels were loaded in `getUnassignedWebsiteCodesForStock`

Reply:
Because service ReplaceSalesChannelsDataForStock work with SalesChannelInterface